wordpress-mobile/WordPress-Android
# Issue/9157 site creation back pressed

Fixes #9157 

~~Note: This PR should be reviewed after #9142 is merged.~~

A warning dialog is shown when the user presses the back button while the site creation is already in progress - the flow is exited when they press the positive dialog button.
The flow is exited right away when they press the back button after the site preview was shown.



![screenshot_1549444610](https://user-images.githubusercontent.com/2261188/52339788-54ac5680-2a0e-11e9-9516-4193b23ff6b2.png)

@SylvesterWilmott I haven't modified the style of the warning dialog to keep it consistent with the rest of the app. Is it ok? Btw has the text been reviewed?

To test:
1. My Site
1. Switch Site
2. Plus sign icon in the top right corner
3. Create a WordPress.com site
1. Make sure the back button works on Segments, Verticals, SiteInfo and Domains screen
2. Go to the domain screen
3. Enter a domain name
2. Click on the "Create Site" button
3. Press the back button while the fullscreen dialog is shown
4. Warning dialog is shown - make sure it survives rotation
5. Click on "Exit"
6. Make sure the flow is dismissed
------------
1. My Site
1. Switch Site
2. Plus sign icon in the top right corner
3. Create a WordPress.com site
4. Select a segment
5. Skip
6. Skip
7. Enter a domain name
1. Click on the "Create Site" button
3. Wait until the site preview is shown
4. ~~Click on the back button and make sure it's suppressed~~ Click on the back button and make sure the flow is exited

Update release notes:

- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
